configuration/cbaSettings.hpp

```diff
@@ -0,0 +1,52 @@
+/* --------------------------------------------------------------------------------------------------------------------
+	Author:	Joecuronium
+	Description:
+		This file allows you to enable different addon settings. Be sure to only enable one per category.
+		If all settings of a category are disabled, you can set those settings manually in the ingame addon menu.
+-------------------------------------------------------------------------------------------------------------------- */
+
+
+//--------------------------------------------------------------------------------------------------------------------------------------
```

Comment:
This file is good and all the options are clear.  I would only suggest a few things:

* Add more documentation at the top to make it clear what this does in-context of the framework (framework relies on editor addon settings, this overrides parts of those settings for convenience, this doesn't replace the importance of the editor settings).
* I think the category requirements should be in the header, as it can be possible to have a multi-enable category in this system.
* The comment separator lines (`//--------------------------------------------------------------------------------------------------------------------------------------`) can be distracting and they also format badly when line-breaks are enabled, so I tend not to favour their use.  I would suggest to use multi-line comments and extended trailing newlines as headers instead.  It's more subtle, but it's less distracting to read:

```
// CA Medical settings (Reopening OFF).
// To enable this, uncomment the line below.
// #define CA_MEDICAL_BASIC 1


/*
    CA2 Downtime Spectator settings

    Be sure to only uncomment one line in this category.
*/

// If this setting is enabled, unconscious players are restricted to 1PP and 3PP spectator modes but dead players have all spectator modes.
// To disable this, comment-out or delete the line below INSTEAD of setting it to 0. 
#define CA2_SPECTATE_LIMITED 1
```